account_invoice_validate_send_email/models/account_invoice.py

```diff
@@ -67,10 +67,16 @@ def action_invoice_open(self):
         base_url = self.env["ir.config_parameter"].get_param("web.base.url")
         for invoice in self:
             term = invoice.payment_term_id
+            order_line = invoice.invoice_line_ids.mapped("sale_line_ids")
+            pickings = self.env["stock.picking"].search(
+                [("sale_id", "in", order_line.order_id)]
+            )
```

Comment:
```suggestion
            orders = invoice.invoice_line_ids.mapped("sale_line_ids").mapped("sale_id")
            pickings = self.env["stock.picking"].search(
                [("sale_id", "in", orders.ids)]
            )
```

Reply:
Thank you, I edit code and it worked as expected.